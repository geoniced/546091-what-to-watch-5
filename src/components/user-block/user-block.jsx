import React from "react";
import {Link} from "react-router-dom";

const UserBlock = () => {
  return (
    <div className="user-block">
      <Link to="/mylist">
        <div className="user-block__avatar">
          <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
        </div>
      </Link>
    </div>
  );
};

export default UserBlock;
