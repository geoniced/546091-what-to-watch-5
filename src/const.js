export const ALL_GENRES_FILTER = `All genres`;
