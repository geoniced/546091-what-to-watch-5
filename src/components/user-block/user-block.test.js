import React from "react";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {AuthorizationStatus} from "../../const";
import {UserBlock} from "./user-block";

describe(`UserBlock render`, () => {
  it(`renders UserBlock component when the user is authorized`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              authorizationStatus={AuthorizationStatus.AUTH}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders UserBlock component when the user is not authorized`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              authorizationStatus={AuthorizationStatus.NO_AUTH}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
