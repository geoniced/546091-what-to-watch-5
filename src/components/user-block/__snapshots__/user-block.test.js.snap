// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock render renders UserBlock component when the user is authorized 1`] = `
<div
  className="user-block"
>
  <a
    href="/mylist"
    onClick={[Function]}
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="img/avatar.jpg"
        width="63"
      />
    </div>
  </a>
</div>
`;

exports[`UserBlock render renders UserBlock component when the user is not authorized 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
